core/src/main/scala/org/apache/spark/storage/BlockManagerId.scala

```diff
@@ -132,10 +133,15 @@ private[spark] object BlockManagerId {
     getCachedBlockManagerId(obj)
   }
 
-  val blockManagerIdCache = new ConcurrentHashMap[BlockManagerId, BlockManagerId]()
+  val blockManagerIdCache = new TimeStampedHashMap[BlockManagerId, BlockManagerId](true)
```

Comment:
Will this cause serious performance regression? `TimeStampedHashMap` will copy all the old values on update. cc @cloud-fan